daml-model/daml/Trigger/LoanManagement.daml

```diff
@@ -0,0 +1,58 @@
+--
+-- Copyright (c) 2022, Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+--
+
+module Trigger.LoanManagement where
+
+import DA.Action (void)
+import DA.Foldable (forA_)
+import DA.Map qualified as Map
+import Daml.Trigger
+import Model.Balance
+import Model.LoanConfig
+
+
+loanManagementTrigger : Trigger ()
+loanManagementTrigger = Trigger
+    { initialize = pure ()
+    , updateState = \_ -> pure ()
+    , rule = run
+    , registeredTemplates = RegisteredTemplates [
+        registeredTemplate @Balance,
+        registeredTemplate @LoanConfig
+        ]
+    , heartbeat = None
+    }
+
+run : Party -> TriggerA () ()
+run sender = do
+    loanConfigs <- query @LoanConfig
+    balances <- query @Balance
+    let balanceMap = Map.fromList $ map ((\x -> ((x.provider, x.iban), x)) . snd) balances
+    forA_ loanConfigs \(loanCid, loanConfig) -> do
+        let maybeBalance = Map.lookup (loanConfig.provider, loanConfig.account) balanceMap
+        case maybeBalance of
+            Some balance -> do
+                createLoanWhenBelowThreshold loanCid loanConfig balance
+                createPaybackWhenAboveThreshold loanCid loanConfig balance
+            _ -> traceA $ "No balance found for " <> loanConfig.account -- TODO: retire LC?
+
+createLoanWhenBelowThreshold : ContractId LoanConfig -> LoanConfig -> Balance -> TriggerA () ()
+createLoanWhenBelowThreshold loanCid loanConfig balance
+    | balance.amount < loanConfig.balanceLoanThreshold = createLoan balance.provider balance.iban None loanCid
+    | otherwise = traceA $ "Maximum loan reached for " <> loanConfig.account
+    where createLoan provider account loanAmount loanConfigCid = do
+            let cmd = exerciseCmd loanConfigCid (CreateLoan with loanAmount = loanAmount)
```

Comment:
not important but you could `let in` and not `do`

Reply:
Here I actually like the `where` :)